msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/Authority.java

```diff
@@ -79,11 +80,11 @@ static AuthorityType detectAuthorityType(URL authorityUrl) {
 
         final String path = authorityUrl.getPath().substring(1);
         if (StringHelper.isBlank(path)) {
-            if(isCiamAuthority(authorityUrl.getHost())){
+            if (isCiamAuthority(authorityUrl.getHost())) {
                 return AuthorityType.CIAM;
+            } else {
+                return AuthorityType.GENERIC;
```

Comment:
I don't think this is correct. You cannot make any assumption on the format of the generic authority. This is why we decided to use a different API "oidcAuthority" - the app developer has to tell MSAL about it. 

if the existing "authority" API is used and MSAL can't detect the authority, then it defaults to AAD.

Reply:
It turns out these changes weren't needed at all: this `createAuthority` method is only used in a few places that had to handle the special multicloud/instance aware scenario, which I don't think is relevant when using the new OidcAuthority API

I reverted the logic back to what it was in the latest commit.